manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml

```diff
@@ -216,6 +216,9 @@ data:
     injectedAnnotations:
 
     template: |
+      {{- $holdProxyPC := boolValueOrDefault .ProxyConfig.HoldApplicationUntilProxyStarts "" }}
```

Comment:
I think this is going to break upgrades and why I recommended we do a different approach (although unfortunately my approach did not work). the old istiod image will not have the `boolValueOrDefault` so it will fail

Reply:
Actually I suppose its the same in minor release or patch release - however, it seems like the pilot image should be more interchangeable for patch releases if possible

Reply:
Oh, good point. I'd not considered that.

I can use a simple 'or' if that would be better. That means that if the global setting is enabled, nobody can disable it.

Reply:
As a side note, can we add Sprig to the injector? I think its reflection commands would have prevented us from needing this.

Reply:
I feel pretty great about my timing on https://github.com/istio/istio/pull/28556 now :slightly_smiling_face:

Reply:
What is the conclusion?  :)  could Sprig also have the upgrade issue as `boolValueOrDefault`?  `or` is certainly the safest.

Reply:
My opinion:

1.8: do `values.global.hold || proxyConfig.hold`. Add deprecation warning for using `values.global.hold`. User impact is no changes if they want to set globally, if they want to override a global disable with pod enable then they can just add the annotation, and if they want to override a global enable with pod disable then they need to switch how they configure the global setting to use mesh config, which they should do anyways.
In master: remove values.global.hold, just use proxyConfig. Everything works as any other proxy config setting

Reply:
I spoke to John offline, it sounds like the best option may be to deprecate the global value in favor of the new one. The template logic will just 'or' the two conditionals, but if someone wants to use the per-pod value they should disable the global one. Then, in master, we can remove the global value entirely.

@linsun would that work for you?

Reply:
That works for me.  We don't have any users that enable values.global.hold atm today.

Thank you @howardjohn and @stewartbutler !

Reply:
cc @luksa since you greatly contributed the values.global.hold in the first place.

Reply:
OK, will tag you all on the next PR unwinding the master change when it is ready.